src/sentry/notifications/utils/participants.py

```diff
@@ -120,16 +110,38 @@ def get_participants_for_release(
     users_to_reasons_by_provider: MutableMapping[
         ExternalProviders, MutableMapping[Team | User, int]
     ] = defaultdict(dict)
+
+    def get_reason(
+        user: User,
+        value: NotificationSettingOptionValues,
+        user_ids: set[int],
+        notification_setting_type: NotificationSettingTypes,
+    ) -> int | None:
+        if notification_setting_type == NotificationSettingTypes.DEPLOY:
+            # Members who opt into all deploy emails.
+            if value == NotificationSettingOptionValues.ALWAYS:
+                return GroupSubscriptionReason.deploy_setting
+
+            # Members which have been seen in the commit log.
+            elif value == NotificationSettingOptionValues.COMMITTED_ONLY and user.id in user_ids:
+                return GroupSubscriptionReason.committed
+        elif notification_setting_type == NotificationSettingTypes.ACTIVE_RELEASE:
+            if value == NotificationSettingOptionValues.ALWAYS:
+                # user must be a committer and a part of this release
+                return GroupSubscriptionReason.committed
+
+        return None
```

Comment:
Moved this method inside this function since `get_reason` was only being called by `get_participants_for_release`. Only downside I see is we can't mock `get_reason()` any more. Not sure if we care to allow that in this scenario. Any strong opinions here?